docs/raw/serving-community-history.md

```diff
@@ -0,0 +1,315 @@
+| -------------------- | --- |
+| Waku node            | An Ethereum node with Waku V1 enabled, or a [10/WAKU2](https://rfc.vac.dev/spec/10/) node that implements [11/WAKU2-RELAY](https://rfc.vac.dev/spec/11/)|
+| Store node           | A Waku node that implements [11/WAKU-MAILSERVER](/spec/11) or [13/WAKU2-STORE](https://rfc.vac.dev/spec/13/) respectively |
+| Waku network         | A group of Waku nodes connected through the internet connection and forming a graph |
+| Community owner      | A Status user that owns a Status community |
+| Community member     | A Status user that is part of a Status community |
+| Community owner node | A Status node with message archive capabilities enabled, run by a community owner |
+| Community member node| A Status node with message archive capabilities enabled, run by a community member |
+| Live messages        | Waku messages received through the Waku network |
+
+
+## Supported platforms
+
+Creating, distributing, downloading and unpacking community history archives SHOULD only be supported in desktop clients. Mobile clients SHOULD NOT implement this functionality due to bandwidth and storage limitations.
+
+## Requirements / Assumptions
+
+This specification has the following assumptions:
+
+- Store nodes are available 24/7, ensuring constant live message availability
+- Store nodes have enough storage to persist historical messages for up to 30 days
+- No store nodes have storage to persist historical messages older than 30 days
+- All nodes are honest
+- The network is reliable
+
+Furthermore, it assumes that:
+
+- Community owner nodes have enough storage to persist historical messages older than 30 days
+- Community owner nodes provide archives with historical messages **at least** every 30 days
+- Community owner nodes receive all community messages
+- Community owner nodes are honest
+
+## Overview
+
+The following is a high-level overview of the user flow and features this specification describes.
+
+### Serving community history archives
+
+Community owner nodes go through the following (high level) process to provide community members with message histories (assumes community owner node is available 24/7):
+
+1. Community owner creates a Status community
+2. Community owner enables community history archive support
+3. A special type of channel for distributing magnet links ([Magnet URI scheme](https://en.wikipedia.org/wiki/Magnet_URI_scheme), [Extensions for Peers to Send Metadata Files](https://www.bittorrent.org/beps/bep_0009.html)) is created
+4. Community owner invites members and creates additional channels
+5. Community owner node receives messages and stores them into local database
+6. After 7 days, the community owner node exports and compresses last 7 days worth of messages from database and creates a magnet link from that data via torrent client
+7. Community owner node creates message archive index and bundles the previously generated magnet link into it
```

Comment:
Yes, as mentioned here https://github.com/status-im/specs/pull/162/files#r767952892, this is technically not the case.

We're not actually bundling the raw data with the new raw data. The reason for that is, we can't guarantee the size in which `pieces` will be created by torrent, so we can't guarantee that the hashes for each `piece` will be the same every time (different length == different hash).

So instead, we're distributing an index that looks like:

```
index: {
  "0x123": {
    from: ...
    to: ...
    magnet_uri: magnet_link_for_archive_1
  },
  "0x456": {
    from: ...
    to: ...
    magnet_uri: magnet_link_to_archive_2
  }
  ...
}
```

Then, based on the hashes (which are derived from the `magnet_uri` which is unique), member nodes can decide which ones they need (all, some, none).

See "Fetching message archives" section for a detailed explanation

Reply:
>  The reason for that is, we can't guarantee the size in which pieces will be created by torrent, so we can't guarantee that the hashes for each piece will be the same every time

Quick update here, this should say "I assume we can't guarantee [...]". 

Torrent clients seem to offer UIs to let users configure the size and/or number of pieces, so it might be possible to go a different route that doesn't use indices.

Will discuss with @staheri14 and update spec accordingly!

Reply:
@PascalPrecht thanks for the call the other evening on this topic.  Yes we definitely should be able set the torrent piece size to a fixed size, irrespective of the size of the torrent.  Prob. better to go for a larger piece size, as this will work better when a torrent grows larger.  4MB seems to be a common torrent piece size